src/JestTestRunner.ts

```diff
@@ -1,131 +1,45 @@
-  collectCoverage: false,
-  haste: {},
-  maxWorkers: 1,
-  moduleFileExtensions: ['js', 'json', 'jsx', 'node'],
-  moduleNameMapper: [],
-  setupFiles: [],
-  snapshotSerializers: [],
-  testEnvironment: 'jest-environment-jsdom',
-  testRegex: '(/__tests__/.*|(\\.|/)(test|spec))\\.jsx?$',
-  testRunner: 'jest-jasmine2',
-  verbose: true
-};
+import JestTestAdapterFactory from './jestTestAdapters/JestTestAdapterFactory';
 
 export default class JestTestRunner extends EventEmitter implements TestRunner {
-  // Seems there are no up-to-date TypeScript definitions for Jest
-
-  private jestAdapter: JestAdapter;
-  private hasteContext: Promise<any>;
-  private options: any;
-  private paths: string[];
+  private jestConfig: any;
+  private projectRoot: string;
 
-  constructor(options: RunnerOptions) {
+  public constructor(options: RunnerOptions) {
     super();
-    this.jestAdapter = findJestAdapter();
-    log.debug(`Received options ${JSON.stringify(options)}`);
 
-    this.options = _.assign(DEFAULT_OPTIONS, {
-      rootDir: process.cwd(),
-      roots: [process.cwd()],
-      testPathDirs: [process.cwd()]
-    });
-    log.debug(`Using options ${JSON.stringify(this.options)}`);
-    
-    const _testRegex = new RegExp(this.options.testRegex);
-    const isTest = (file: string) => _testRegex.test(file);
+    this.projectRoot = process.cwd();
 
-    this.paths = options.files
-      .map(file => file.name)
-      .filter(isTest);
-
-    log.info(`Discovered specs ${JSON.stringify(this.paths)}`);
-  }
-
-  init(): Promise<any> | void {
-    log.info(`Initializing Jest`);
-    this.hasteContext = this.jestAdapter.buildHasteContext(this.options);
+    this.jestConfig = options.strykerOptions.jest.config;
+    this.jestConfig.rootDir = this.projectRoot;
   }
 
-  run(): Promise<RunResult> {
-    return this.hasteContext
-      .then((context: any) => this.runTests(context))
-      .then((results: any[]) => this.processJestResults(results))
-      .catch((error: Error) => this.catchError(error));
-  }
+  public async run(): Promise<RunResult> {
+    const jestTestRunner = JestTestAdapterFactory.getJestTestAdapter();
 
-  private runTests(hasteContext: any): Promise<any[]> {
-    const promises = this.paths.map((specPath: string) => {
-      return this.jestAdapter.runTest(
-        path.resolve(specPath),
-        this.options,
-        hasteContext.resolver
-      );
-    });
-    return Promise.all(promises);
-  }
+    const { results } = await jestTestRunner.run(this.jestConfig, process.cwd());
 
-  private catchError(error: Error): RunResult {
-    log.error(`An error occurred while invoking Jest: ${error.stack}`);
     return {
-      coverage: undefined,
-      tests: [],
-      status: RunStatus.Error,
-      errorMessages: [ `${error.name}: ${error.message}` ]
+      tests: this.processTestResults(results.testResults),
+      status: (results.numRuntimeErrorTestSuites > 0) ? RunStatus.Error : RunStatus.Complete
```

Comment:
Is there a way to get at the actual error messages? How would people start to debug when there is a runtime error?

Reply:
It only returns a failureMessage for the individual test suites, not for the run as a whole.

Reply:
are there no runtime error messages on the run result or test file result? Seems strange to me that they do show the number of runtime errors, but not the errors themselves.

Reply:
They are on the fileResult, but there are more fileResults in a single runResult.